ado/ipacheckreadreplace.ado

```diff
@@ -44,31 +44,41 @@ qui {
 			error 110
 		}
 	}
-	* import corrections dataset
+	****** import corrections dataset and check for fatal errors ***************
 	preserve
 	if missing("`use'") & missing("`excel'") {
 		local insheet insheet
-		import delimited `using', `import' clear
+		import delimited using "`using'", `import' clear
 	}
 	else if !missing("`excel'") {
-		import excel `using', `import' clear
+		import excel using "`using'", `import' firstrow clear
 	}
 	else if !missing("`use'") {
-		local using subinstr("`using'", "using", "", 1)
-		use `using', `import' clear
+		use "`using'", `import' clear
 	}
+	if _N == 0 {
+		di "no replacements"
+		exit
+	}
+	
+	cap assert !missing(`id')
```

Comment:
This is before we hit the readreplace command, and it makes sure that the form has what it needs (ie, nonmissing ids) for the "drop" section and the "select multiple" section as well.